packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-point-settings.tsx

```diff
@@ -77,8 +127,54 @@ const LockedPointSettings = (props: Props) => {
                 onChange={(newValue) => {
                     onChangeProps({filled: !newValue});
                 }}
+                style={styles.spaceUnder}
             />
 
+            {flags?.["mafs"]?.["locked-point-labels"] && (
+                <>
+                    {labels.map((label, labelIndex) => (
+                        <LockedLabelSettings
+                            {...label}
+                            expanded={true}
+                            onChangeProps={(newLabel: LockedLabelType) => {
+                                handleLabelChange(newLabel, labelIndex);
+                            }}
+                            onRemove={() => {
+                                handleLabelRemove(labelIndex);
+                            }}
+                        />
+                    ))}
+
+                    <Button
+                        kind="tertiary"
+                        startIcon={plusCircle}
+                        onClick={() => {
+                            const newLabel = {
+                                ...getDefaultFigureForType("label"),
+                                // Place the label at the same position
+                                // as the point. Add a small offset to
+                                // avoid crowding.
+                                coord: [
+                                    coord[0] + 0.5,
+                                    // Additional offset for each label so
+                                    // they don't overlap.
+                                    coord[1] - 1 * labels?.length,
```

Comment:
These coords are in graph units, right? I think these offsets might not work well for graphs that have ranges other than `[[-10, 10], [-10, 10]]`. I guess that's okay, though, because these values are just defaults?

Reply:
Hmmm... that's a good point. I'm not sure how much it would be worth calculating the offset, since I suspect authors would change the position in most cases anyway? I guess we could get feedback from them and then update accordingly in the future.